### PR TITLE
bootstrap carousel v1.0

### PR DESCRIPTION
**Pull request by cailors**
- RECOMMANDATION_STANDARD: number limit to be showed up in a carousel
- filters and search inputs  DOES NOT affect the carousel

Screen Shot 2022-08-04 at 6 25 42 PM

Screen Shot 2022-08-04 at 6 24 59 PM

### CODE REVIEW

File: frontend/src/components/my-items/AddItemForm.jsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ export default function AddItemForm (props) {
     formData.append('description', description)
     formData.append('status', switchIsAvailable ? STATUS.AVAILABLE : STATUS.NOT_AVAILABLE)
     formData.append('image', fileInput.current.files[0])
+    formData.append('numberOfTimesBorrowed', 0)
```
Comment (imogenem) [inline frontend/src/components/my-items/AddItemForm.jsx]: I don't thiiiink you need to set this here, instead you can set a default in the Item model. 

```
ie
  numberOfTimesBorrowed: {
    type: Number,
    default: 0
  },
```

File: frontend/src/components/marketplace/MarketplacePage.jsx [hunk 2]
```diff
@@ -24,6 +25,10 @@ function MarketplacePage () {
       <div className="grid-container">
         <div className="grid-child page-container" key = "grid-child page-container">
           <FiltersCollection/>
+          <br></br>
+          <br></br>
+          <br></br>
+          <ControlledCarousel items={items}/>
```
Comment (imogenem) [inline frontend/src/components/marketplace/MarketplacePage.jsx]: To make it distinct (in it's own container with a grey background and border), you can adjust the grid. So the grid would have two small sections on the left.

something like
```
<div grid-child>
<FiltersCollection />
</div>
<div grid-child>
<ControlledCarousel />
</div>
```

File: frontend/src/components/marketplace/ControlledCarousel.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      id={item._id}
+      borrow
+      description={item.description}
+      image={item.image}
+      name={item.name}
+      status={item.status}
+      ratingComments={item.ratingComments}
+      type={item.type}
+      rating={item.rating}
+      location={item.owner.location}/>
```
Comment (imogenem) [inline frontend/src/components/marketplace/ControlledCarousel.jsx]: We should show the number of times borrowed on the card so people understand why it's there, just need to pass in the `numberOfTimesBorrowed` prop. That should help with local testing too!

File: frontend/src/components/marketplace/ControlledCarousel.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  const handleSelect = (selectedIndex, e) => {
+    setIndex(selectedIndex)
+  }
+
+  const items = prop.items
+
+  const itemList = items.map((item) => {
+    if (!item.numberOfTimesBorrowed || item.numberOfTimesBorrowed < RECOMMANDATION_STANDARD) {
+      return null
+    }
```
Comment (imogenem) [inline frontend/src/components/marketplace/ControlledCarousel.jsx]: Add a title to this component so we know what it is

File: frontend/src/components/marketplace/ControlledCarousel.jsx (hunk 3 above)
Comment (cailors) [reply frontend/src/components/marketplace/ControlledCarousel.jsx]: I decided to do this just for testing purpose: description={item.numberOfTimesBorrowed}
I don't think it it worth it to add it to the ItemCard.

Comment (imogenem) [reply frontend/src/components/marketplace/ControlledCarousel.jsx]: @cailors could we add it conditionally just to the featured cards on the carousel so people understand why the cards are there?

**Comment by imogenem** [comment]
Updated page looks good, but it looks like the cards should have a max-width so it doesn't stretch to fill the entire container (if there is just one card in the carousel).

File: frontend/src/components/marketplace/ControlledCarousel.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  }, [dispatch])
+
+  const displayedInCarouselItems = items.map((item) => {
+    if (item.numberOfTimesBorrowed == null || item.numberOfTimesBorrowed < RECOMMANDATION_STANDARD) {
+      return null
+    }
+    return <Carousel.Item><ItemCard key={item._id}
+      id={item._id}
+      borrow
+      description={item.numberOfTimesBorrowed}
```
Comment (imogenem) [inline frontend/src/components/marketplace/ControlledCarousel.jsx]: change back to displaying the description before merging this to dev

**Comment by cailors** [comment]
> Updated page looks good, but it looks like the cards should have a max-width so it doesn't stretch to fill the entire container (if there is just one card in the carousel).

Fixed it a bit but img will still stretch out a bit while the screen size is between full screen and "mobile screen". Not sure if there are better solution:(

File: frontend/src/components/marketplace/ControlledCarousel.jsx (hunk 4 above)
Comment (phreid) [reply frontend/src/components/marketplace/ControlledCarousel.jsx]: this is still missing

**Comment by imogenem** [comment]
Updated version

Screen Shot 2022-08-05 at 4 06 49 PM

**Comment by cailors** [comment]
Unfortunately, have to remove the text. Right now fixing the alignment of the item card
Screen Shot 2022-08-05 at 7 36 32 PM

**Comment by cailors** [comment]
Screen Shot 2022-08-06 at 2 47 10 PM
Able to scroll when screen size shrinks
Screen Shot 2022-08-06 at 2 47 35 PM
pending, to be fixed:
Screen Shot 2022-08-06 at 2 49 12 PM

**Comment by cailors** [comment]
I decided to go easy with it, no carousel when there is no recommended item
Screen Shot 2022-08-06 at 5 59 56 PM
The text will be the first carousel item
Screen Shot 2022-08-06 at 6 03 42 PM
Screen Shot 2022-08-06 at 6 03 52 PM
Able to scroll when screen size shrinks
Screen Shot 2022-08-06 at 2 47 35 PM

File: frontend/src/components/marketplace/MarketplacePage.jsx (hunk 2 above)
Comment (cailors) [reply frontend/src/components/marketplace/MarketplacePage.jsx]: used className='row' and className='col' instead

**Review by nyananu — APPROVED** [review]

**Pull request merged**